herbert_core/utilities/general/is_file.m

```diff
@@ -0,0 +1,45 @@
+function ok = is_file(varargin)
+% tests if name is a file on the file system while NOT searching MATLAB path
+%
+% Optional support for a cell array of file extensions which name must also match.
+%
+% On older versions of matlab this is done through ensuring the path is an explicit path
+% and using exist. More recent versions simply call the MATLAB built-in isfile.
+%
+% Input:
+% ------
+%   name                The name of the file you want to check;
+%   {'.a','.b','.c'}    List of the permitted extensions
```

Comment:
I'm not sure if I understand what these permitted extensions are for? Is the following the correct behaviour, or am I misunderstanding?

```matlab
>> ls

.                      test_cut.m             test_duplicated_id.m   test_sqw_eval.m        
..                     test_cut_ref_sqw.sqw   test_func_eval.m       test_sqw_eval_ref.sqw  

>> is_file('test_sqw_eval_ref.sqw')

ans =

  logical

   1

>> is_file('test_sqw_eval_ref')

ans =

  logical

   0

>> is_file('test_sqw_eval_ref', {'.sqw'})

ans =

  logical

   0
```

Reply:
This was originally intended to reproduce the functionality of [check_file_exist](https://github.com/pace-neutrons/Herbert/blob/master/herbert_core/classes/data_loaders/check_file_exist.m). But I had misunderstood its operation